12-version_control.Rmd

```diff
@@ -0,0 +1,821 @@
+image_read("img/version_control/clone_02.png") %>%
+  image_crop("3584x1500")
+```
+
+Then open JupyterHub, and click the Git+ icon on the file browser tab 
+(Figure \@ref(fig:clone-01)).
+
+```{r clone-01, fig.cap = 'The Jupyter Git Clone icon (red circle)', fig.retina = 2}
+image_read("img/version_control/clone_01.png") %>%
+  image_crop("2400x1430+1")
+```
+
+Paste the URL of the GitHub project repository you 
+created and click the blue "CLONE" button (Figure \@ref(fig:clone-03)).
+
+```{r clone-03, fig.cap = 'Prompt where the remote address (URL) corresponding to the location of the GitHub repository needs to be input in Jupyter.', fig.retina = 2}
+image_read("img/version_control/clone_03.png") %>%
+  image_crop("2400x1430+1")
+```
+
+On the file browser tab, you will now see a folder for your project's repository 
+(and inside it will be all the files that existed on GitHub, 
+Figure \@ref(fig:clone-04)).
+
+```{r clone-04, fig.cap = 'Cloned GitHub repositories can been seen and accessed via the Jupyter file browser.', fig.retina = 2}
+image_read("img/version_control/clone_04.png") %>%
+  image_crop("2400x1430+1")
+```
+
+## Working in a cloned repository using Jupyter
+
+Now that you have cloned your repository from GitHub to your local computer
+(*i.e.,* JupyterHub or your laptop), you can get to work editing, creating, 
+and deleting files using an application, such as Jupyter. 
+Below we walk through how to use the Jupyter Git extension tool to do 
+each of the steps outlined above.
+
+### Specifying files to commit
+
+Below we created and saved a new file (named `eda.ipynb`) that we would 
+like to send back to the project repository on GitHub.
+To "add" this modified file to the staging area (*i.e.*, flag that this is a
+file whose changes we would like to commit), we click the Jupyter Git extension 
+icon on the far left-hand side of JupyterLab (Figure \@ref(fig:git-add-01)).
+
+```{r git-add-01, fig.cap = 'Jupyter Git extension icon (circled in red).', fig.retina = 2}
+image_read("img/version_control/git_add_01.png")
+```
+
+This opens the Jupyter Git graphical user interface pane, and then we
+click the plus sign (+) beside the file that we want to "add" 
+(Figure \@ref(fig:git-add-02)). 
+
+> **Note:** because this is the first change for this file that we want to add, 
+> it falls under the "Untracked" heading. 
+> However, next time we edit this file 
+> and want to add the changes we made, 
+> we will find it under the "Changed" heading.
+
+> **Note:** do not add the `eda-checkpoint.ipynb` file 
+> (sometimes called `.ipynb_checkpoints`). 
+> This file is automatically created by Jupyter when you work on `eda.ipynb`. 
+> You generally do not add auto-generated files to Git repositories,
+> only add the files you directly create and edit.
+
+```{r git-add-02, fig.cap = 'The plus sign (+) is used to add the `eda.ipynb` file to the staging area.', fig.retina = 2}
```

Comment:
I think this was a comment that Trevor brought up about not using our real github user account names.... Though I understand it may be quite bothersome to redo all these screenshots...

Reply:
Right... I will add this to the version control chapter final touches so we get the most recent version of the GitHub web app UI before publishing.